busy_beaver/apps/github_summary/workflow.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import uuid
+from urllib.parse import urlencode
+
+from busy_beaver.config import GITHUB_CLIENT_ID, GITHUB_REDIRECT_URI, MEETUP_GROUP_NAME
+from busy_beaver.extensions import db
+from busy_beaver.models import User
+from busy_beaver.toolbox import EventEmitter
+
+#Setup logging
+logger = logging.getLogger(__name__)
+slash_command_dispatcher = EventEmitter()
+
+#String variables to describe state
+ACCOUNT_ALREADY_ASSOCIATED = (
+    "You have already associated a GitHub account with your Slack handle. "
+    "Please use `/busybeaver reconnect` to link to a different account."
+)
+NO_ASSOCIATED_ACCOUNT = (
+    "No associated account. Use `/busybeaver connect` to link your account."
+)
+VERIFY_ACCOUNT = (
+    "Follow the link below to validate your GitHub account. "
+    "I'll reference your GitHub username to track your public activity."
+)
+
+DELETE_ACCOUNT = (
+    "Account has been deleted. Run '/busybeaver connect' to reconnect"
+)
+
+#functions to create, delete, and manipulate github attachments.
+def add_tracking_identifer_and_save_record(user: User) -> None:
+    user.github_state = str(uuid.uuid4())  # generate unique identifer to track user
+    db.session.add(user)
+    db.session.commit()
+    return user
+
+
+def create_github_account_attachment(state):
+    data = {
+        "client_id": GITHUB_CLIENT_ID,
+        "redirect_uri": GITHUB_REDIRECT_URI,
+        "state": state,
+    }
+
+    query_params = urlencode(data)
+    url = f"https://github.com/login/oauth/authorize?{query_params}"
+    return {
+        "fallback": url,
+        "attachment_type": "default",
+        "actions": [{"text": "Associate GitHub Profile", "type": "button", "url": url}],
+    }
+
+def check_account_existing(slack_id):
+    user_record = User.query.filter_by(slack_id=slack_id).first()
+
+    if user_record:
+        logger.info("[Busy Beaver] Slack account already linked to Github")
+        return True
+
+    logger.info("[Busy Beaver] New user. No Github account associated with profile.")
+    return False
+
+
+def generate_account_attachment(**data):
```

Comment:
I'm not sure what an `account_attachment` is; `create_or_update_account` describes what we are doing here

Reply:
I used attachment to describe the function because busy beaver's not creating or updating a github account, we're associating/attaching the account with busy beaver. Though that maybe my misunderstanding, what would you advise?